bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,269 @@
+            }
+        }
+    }
+
+    private void checkStatus() {
+        String url = "http://" + config.ip + "/statusn";
+
+        try {
+            ContentResponse resp = httpClient.newRequest(url).method(HttpMethod.GET).send();
+            final String response = resp.getContentAsString();
+            logger.debug("status values: {}", response);
+            if (resp.getStatus() == 200) {
+                processStatusResponse(response);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "got response code: " + resp.getStatus());
+            }
+            // reconnect web socket if not connected
+            if (config.useWebSockets && (session == null || !session.isOpen())
+                    && ThingStatus.ONLINE == getThing().getStatus()) {
+                logger.debug("Initializing WebSocket session");
+                initializeWebSocketSession();
+                return;
+            }
+        } catch (InterruptedException e) {
+            logger.debug("API execution has been interrupted", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "API execution has been interrupted");
+            Thread.currentThread().interrupt();
+        } catch (TimeoutException e) {
+            logger.debug("Timeout during status update", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Timeout during status update");
+        } catch (ExecutionException e) {
+            logger.debug("Exception during status update", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Exception during status update: " + e.getMessage());
+        }
+        session = null;
+    }
+
+    public synchronized void processStatusResponse(String content) {
+        if (ThingStatus.ONLINE != getThing().getStatus()) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        Map<String, String> values = Stream.of(content.split("\n")).map(s -> s.split(":"))
+                .filter(s -> s.length == 2).collect(Collectors.toMap(s -> s[0], s -> s[1]));
+
+        updateProperty("Product type", values.getOrDefault("t", NA));
+        updateProperty(Thing.PROPERTY_FIRMWARE_VERSION, values.getOrDefault("fw", NA));
+        updateProperty("Wifi signal", values.getOrDefault("rssi", NA));
+        updateProperty("Product name", values.getOrDefault("cid", NA));
+
+        String statusMessage = "Stopped";
+        if (!"250".equals(values.getOrDefault("s", NA))) {
+            statusMessage = "Moving";
+        }
+
+        if ("1".equals(values.getOrDefault("ro", NA))) {
+            statusMessage = "Not ready";
+        }
+
+        if ("10".equals(values.getOrDefault("e", NA))) {
+            statusMessage = "Blocked";
+        }
+
+        int position = values.getOrDefault("pos", NA).equals("o") ? 0 : 100;
+        if (position == 0 && lastPosition != -1) {
```

Comment:
I do not understand this. If you receive a position, why can't you use that one? Or - if "o"  indicates an invalid position, why don't you just skip the channel update? That way you could remove the whole `lastPosition` handling (which would be great, because we try to not cache states).

Reply:
the only position the device reports is "o" = OPEN and "c" = CLOSED. But you can send the device to a percentage position using DIMMER. In that case it reports "o" and it's quite handy to have this position so it is visible the window position is between OPEN and CLOSED.